pkg/kclient/ingress.go

```diff
@@ -22,17 +24,55 @@ func (c *Client) CreateIngress(ingress extensionsv1.Ingress) (*extensionsv1.Ingr
 	return ingressObj, nil
 }
 
-// DeleteIngress deletes the given ingress
-func (c *Client) DeleteIngress(name string) error {
+func (c *Client) CreateNetworkingV1Beta1Ingress(ingress networkingv1beta1.Ingress) (*networkingv1beta1.Ingress, error) {
+	if ingress.GetName() == "" {
+		return nil, fmt.Errorf("ingress name is empty")
+	}
+	ingressObj, err := c.KubeClient.NetworkingV1beta1().Ingresses(c.Namespace).Create(context.TODO(), &ingress, metav1.CreateOptions{FieldManager: FieldManager})
```

Comment:
I was just thinking if we can create an interface for functions we care about from networking side and then pass that around to reduce code duplication?

Reply:
I guess that could be possible. But it might be hard as we will need to have a common interface object for that to work. Create for eg takes an ingress object